### PR TITLE
Add support for musl linux distros

### LINKED ISSUE #7: Build for Alpine linux?
I often use the `-alpine` images for Python in docker, but there doesn't seem to be a build compiled against `musl` so it requires me to install it into my container.

What would be required for a build that can install into an alpine docker image ?

I'm currently reading through https://github.com/PyO3/maturin-action?tab=readme-ov-file#examples, to see if there are any examples that might apply here.

My first thought was that this could be converted to a platform matrix
https://github.com/messense/py-promql-parser/blob/d86aeb6b88656a5056d3150909bb5a5b47a1577e/.github/workflows/CI.yml#L12-L25

perhaps something similar to here
https://github.com/pydantic/pydantic-core/blob/b777774d0fa73a400152c31c33c85fb824e59d7e/.github/workflows/ci.yml#L399-L434

where the `musllinux` builds can be custom added to the `include` in case the matrix build setting doesn't work properly.

Is there any guidance you can provide ?

### PR DESCRIPTION
**Pull request by kfdm**
This adapts the workflow configuration from [1] and applies it here, so that we can build promql-parser packages for distrobutions that use musl.
    
[1] https://github.com/messense/nh3/blob/b5074b186b813313b258a7c97871bb2d9fc0eaa7/.github/workflows/CI.yml

Fixes #7

### CODE REVIEW

**Comment by kfdm** [comment]
I'm still trying to test on my fork, though I have not yet gotten `x86_64-unknown-linux-musl` to build correctly. Posting here for early feedback. 

https://github.com/kfdm/py-promql-parser/pull/1

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -24,6 +28,82 @@ jobs:
...
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+          architecture: x64
+      - name: Build wheels
+        uses: messense/maturin-action@v1
+        with:
+          target: ${{ matrix.target }}
+          manylinux: musllinux_1_2
+          args: --release --out dist --find-interpreter
```
Comment (messense) [inline .github/workflows/CI.yml]:
```suggestion
          args: --release --out dist -i 3.7 3.8 3.9 3.10 3.11 3.12
```

seems like a version issue, it only built for 3.10 before.

**Comment by kfdm** [comment]
Thank you for the pointer! Looks like that fixed it in my fork so I'll commit your suggestion and let it run again.

**Review by messense — APPROVED** [review]

**Comment by messense** [comment]
Will push a new release after work.

**Comment by kfdm** [comment]
No rush! Thanks again for the python bindings and guiding me through adding support 🙇

**Comment by messense** [comment]
Released in v0.4.0.

**Pull request merged**